integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java

```diff
@@ -749,4 +754,75 @@ public static void shutdownDomainAndVerify(String domainNamespace, String domain
+   *                      ConfigChangesPendingRestart
+   * @return true if the condition type does not exist, false otherwise
+   */
+  public static boolean verifyDomainStatusConditionTypeDoesNotExist(String domainUid,
+                                                                    String domainNamespace,
+                                                                    String conditionType) {
+    return verifyDomainStatusConditionTypeDoesNotExist(domainUid, domainNamespace,
+            conditionType, DOMAIN_VERSION);
+  }
+
+  /**
+   * Check the domain status condition type does not exist.
+   * @param domainUid uid of the domain
+   * @param domainNamespace namespace of the domain
+   * @param conditionType the type name of condition, accepted value: Completed, Available, Failed and
+   *                      ConfigChangesPendingRestart
+   * @param domainVersion version of domain
+   * @return true if the condition type does not exist, false otherwise
+   */
+  public static boolean verifyDomainStatusConditionTypeDoesNotExist(String domainUid,
+                                                                    String domainNamespace,
+                                                                    String conditionType,
+                                                                    String domainVersion) {
+    Domain domain = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace,
+            domainVersion));
+
+    if (domain != null && domain.getStatus() != null) {
+      List<DomainCondition> domainConditionList = domain.getStatus().getConditions();
+      for (DomainCondition domainCondition : domainConditionList) {
+        if (domainCondition.getType().equalsIgnoreCase(conditionType)) {
+          return false;
+        }
+      }
+    } else {
+      if (domain == null) {
+        getLogger().info("domain is null");
+      } else {
+        getLogger().info("domain status is null");
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Obtains the specified domain, validates that it has a spec and no rolling condition.
+   * @param domainNamespace the namespace
+   * @param domainUid the UID
+   */
+  @org.jetbrains.annotations.NotNull
+  public static Domain getAndValidateInitialDomain(String domainNamespace, String domainUid) {
+    Domain domain = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+            String.format("getDomainCustomResource failed "
+                            + "with ApiException when tried to get domain %s in namespace %s",
+                    domainUid, domainNamespace));
+
+    assertNotNull(domain, "Got null domain resource");
+    assertNotNull(domain.getSpec(), domain + "/spec is null");
+    assertFalse(domainHasRollingCondition(domain), "Found rolling condition at start of test");
+    return domain;
+  }
+
+  private static boolean domainHasRollingCondition(Domain domain) {
```

Comment:
I believe these conditions doesn't exist in 3.4. It will not make the tests to fail, so its fine.

Reply:
Let's remove the assertion. It's strange to assert for the lack of a condition that doesn't exist in 3.4.